src/commons/stores/brick-store.js

```diff
@@ -45,6 +45,53 @@
                 return obj;
             }
         };
+
+        store.glusterBrickMapping = function(cluster, selectedSubVolume) {
+            var deferred,
+                postData;
+
+            deferred = $q.defer();
+            postData = _generatePostDataForBrickLayout();
+            utils.glusterBrickMapping(postData, cluster)
+                .then(function(data) {
+                    deferred.resolve(data);
+                });
+
+            return deferred.promise;
+
+            function _generatePostDataForBrickLayout(){
+                var bricksPostData = {},
+                    nodesForBricks = {},
+                    allNodes = cluster.nodes,
+                    node,
+                    totalBricksForCreatingVolume = 0,
+                    nodeWithMinBrick = 0;
+
+                for (node in allNodes) {
+                    if(allNodes[node].hostCheckBoxSelected === true) {
+                        nodesForBricks[node] = null;
+                        totalBricksForCreatingVolume += allNodes[node].bricksAvaialble;
+                    }
+                }
+                bricksPostData["Cluster.node_configuration"] = nodesForBricks;
+                bricksPostData["Volume.brick_count"] = totalBricksForCreatingVolume;
+                bricksPostData["Volume.subvol_size"] = selectedSubVolume;
+                return bricksPostData;
+            }
+        };
```

Comment:
@cloudbehl No need of semicolon here. Its funtion declaration not function expression